### PR TITLE
Remove JSHint since we're using ESLint in commit hooks

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Since we're using ESLint in our commit hooks, there's no reason to keep the JSHint configuration and tasks around. My IDE picked it up automatically and the two don't seem to agree on style. This PR removes the `.jshintrc` file and replaces the Grunt task for `jshint` with an `eslint` task. I've also removed the linting from the build process for now, as it'll cause a failure due to errors. This should be fine because our pre-commit hooks run eslint, so any changes in the file should result in that file becoming compliant (assuming the developer fixes the errors 😁)

### How to test the changes in this Pull Request:

1. Make sure you update your `node_modules` after checking out the PR!
2. Run `npm run build` and make sure there's no failure caused by ESLint
3. Run `grunt eslint` and see the output, should run ESLint

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -197,7 +193,7 @@ module.exports = function( grunt ) {
 					'!<%= dirs.js %>/admin/*.min.js',
 					'!<%= dirs.js %>/frontend/*.min.js'
 				],
-				tasks: ['jshint', 'uglify']
+				tasks: ['uglify']
```
Comment (claudiosanches) [inline Gruntfile.js:196]: No more lint while converting files? Maybe allow only warnings?

File: Gruntfile.js [hunk 2]
```diff
@@ -291,7 +287,7 @@ module.exports = function( grunt ) {
 	grunt.loadNpmTasks( 'grunt-rtlcss' );
 	grunt.loadNpmTasks( 'grunt-postcss' );
 	grunt.loadNpmTasks( 'grunt-stylelint' );
-	grunt.loadNpmTasks( 'grunt-contrib-jshint' );
+	grunt.loadNpmTasks("gruntify-eslint");
```
Comment (claudiosanches) [inline Gruntfile.js:290]: Coding standards...

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Needs at least fix coding standards in the Gruntfile.js.

File: Gruntfile.js (hunk 2 above)
Comment (ObliviousHarmony) [reply Gruntfile.js:290]: The joys of copy/paste an an over-reliance on automated fixing tools 😆 

I guess Gruntfile.js isn't checked, I'll fix that too!

File: Gruntfile.js (hunk 1 above)
Comment (ObliviousHarmony) [reply Gruntfile.js]: *grumble grumble* I fixed all of the ESLint errors 😫

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Need to fix a conflict with `package.json`

**Comment by ObliviousHarmony** [comment]
Resolved, save us from further merge conflicts!

**Review by claudiosanches — APPROVED** [review]
Works great now!

**Pull request merged**